internal/controller/util.go

```diff
@@ -138,3 +139,38 @@ func isStableState(ctx context.Context, c client.Client, pod *corev1.Pod) (bool,
 
 	return true, nil
 }
+
+func redisClientFromPod(pod *corev1.Pod) *redis.Client {
+	opts := &redis.Options{
+		Addr: fmt.Sprintf("%s:%d", pod.Status.PodIP, 6379),
+	}
+	if password, found := searchForPassword(pod); found {
+		opts.Password = password
+	}
+	return redis.NewClient(opts)
+}
+
+// searchForPassword will search the pod spec for the redis password.
+// The password can be set in two ways:
+// 1. The env variable PasswordEnvironmentKey
+// 2. The argument PasswordArgumentKey
+// For each container, searchForPassword will look at the environment first, then the arguments,
+// and it will return on the first match it finds.
+func searchForPassword(pod *corev1.Pod) (string, bool) {
+	for _, container := range pod.Spec.Containers {
```

Comment:
It doesn't make sense to search for the password in every container. Though we don't have sidecar containers for now, we should limit the scope to dragonfly containers only.

Add a `container.Name == "dragonfly"` check so that we only look into the dragonfly container env vars.

Reply:
We don't need this function anymore as we are now using admin-port to prevent password or tls-enabled authentication to manage drangonfly instances by the operator.

Reply:
Awesome! That is a much better solution. I'll remove this.